transport/wseb/src/main/java/org/kaazing/gateway/transport/wseb/WsebSession.java

```diff
@@ -423,13 +455,24 @@ private void attachReader0(final IoSessionEx newReader) {
         // TODO: needs re-alignment similar to attachWriter
         if (newReader instanceof HttpAcceptSession) {
             HttpAcceptSession newAcceptReader = (HttpAcceptSession) newReader;
-            if (isReaderOutOfOrder(newAcceptReader)) {
-                closeSession(newAcceptReader);
+            if (!checkReaderOrder(newAcceptReader)) {
                 return;
             }
         }
 
         readerSequenceNo++;
+        IoSessionEx oldReader = readSession.get();
+        if (oldReader != null && !oldReader.isClosing() && oldReader instanceof HttpAcceptSession) {
+            // Overlapping upstream, forbidden
+            String message = String.format("Overlapping upstream request");
```

Comment:
Yes, WsebAcceptor doesn't call detachReader for some reason. If we don't do that, my guess is that it may conditionally satisfy some requests as overlapping upstream requests.